### ISSUE TITLE
Error with SetupCasperAPI.py

### ISSUE THREAD

**Issue by benducklow** [eventbrite/Casper-API-Tools#1]
Forgive me up front, but I am a python new. Running El Cap and I have local admin rights. I would very much like to use this script/toll of your but I am getting the following error when trying to set things up:

```
$ python SetupCasperAPI.py
Welcome to the Casper CLI setup. Please note that this will overwrite any configuration files you have previously configured.

Enter JSS URL in the form yourjss.jamfcloud.com: CASPERSERVER.domain.com
Username: admin	
Password: 
Enter full path to the location to store encryption keys, in the form /Path/To/Folder : /Volumes/Keys
unable to write 'random state'
unable to write 'random state'
```

It looks like it a permissions issue with my /Volumes/Keys? I am able to create a blank file and folder ok via the GUI... Please let me know if you may know what is happening. Thanks!

**Comment by jkuo-eb** [comment]
Hi Ben - can you try running that as `sudo python SetupCasperAPI.py`?

**Comment by benducklow** [comment]
Thanks for the response! That got me through those two errors, however when I try to simply run the `casper get computer ben` (for example) command I get this:
```
$ casper get computer bducklo
Traceback (most recent call last):
  File "/Volumes/CasperAPI/CasperAPI_CLI.py", line 1523, in <module>
    main()
  File "/Volumes/CasperAPI/CasperAPI_CLI.py", line 1446, in main
    getComputer(compSearch, user, password, detail)
  File "/Volumes/CasperAPI/CasperAPI_CLI.py", line 269, in getComputer
    r = sendAPIRequest(reqStr, username, password, 'GET')
  File "/Volumes/CasperAPI/CasperAPI_CLI.py", line 88, in sendAPIRequest
    response = urllib2.urlopen(request)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 154, in url open
    return opener.open(url, data, timeout)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 431, in open
    response = self._open(req, data)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 449, in _open
    '_open', req)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 409, in _call_chain
    result = func(*args)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 1240, in https_open
    context=self._context)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 1197, in do_open
    raise URLError(err)
urllib2.URLError: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:590)>
```
I've tried putting in the ":8443" prefix in the jssUrl section with no change. Can't seem to do this against my DEV or PROD JSSs. Note, I host the JSSs onsite.

**Comment by jkuo-eb** [comment]
Ben - since you set up the script using `sudo`, can you try changing your bash_profile to the following instead?

`alias casper="sudo python /Volumes/CasperAPI/CasperAPI_CLI.py"`

Then reload your bash profile by typing:

`source ~/.bash_profile`

and try the command again, like `casper getcomputer ben` ?

The other thing to look into is the server cert, since it's internally hosted. I've only tested this on JAMF Cloud, so that'll take some more digging.

**Comment by benducklow** [comment]
OK, I think there were a few things going on. 

1. I had an Unverified MDM Profile on my admin machine (long story). To fix, I had to remove MDM and re-install MDM on it (its managed by Casper as well). Not sure if this was the issue, but it was something I did today.
2. I had to enter the port # as part of the JSS url during the `$ sudo python SetupCasperAPI.py` setup (ex: CASPERSERVER.domain.com:8443). I also did add the `alias casper="sudo python /Volumes/CasperAPI/CasperAPI_CLI.py"` to my bash profile.

My simple computer name lookup test was successful! Now, I just need to find some time to start playing with this more! I really would love to feed it a *.csv of machine name to get them set to Unmanaged. Thanks much!!
